src/stripeLinter.ts

```diff
@@ -73,33 +75,30 @@ export const shouldSearchFile = (currentFilename: string): boolean => {
 
 export class StripeLinter {
   telemetry: Telemetry;
+  git: Git;
 
-  constructor(telemetry: Telemetry) {
+  constructor(telemetry: Telemetry, git: Git) {
     this.telemetry = telemetry;
+    this.git = git;
   }
 
-  activate() {
-    this.lookForHardCodedAPIKeys();
+  async activate() {
+    if (window.activeTextEditor) {
+      await this.lookForHardCodedAPIKeys(window.activeTextEditor.document);
+    }
     workspace.onDidSaveTextDocument(this.lookForHardCodedAPIKeys);
   }
 
-  lookForHardCodedAPIKeys = (): void => {
-    const editor = window.activeTextEditor;
-    if (!editor) { return; }
-
-    const {document} = editor;
-
+  lookForHardCodedAPIKeys = async (document: TextDocument): Promise<void> => {
     const currentFilename = document.uri.path;
     if (!shouldSearchFile(currentFilename)) { return; }
 
     const text = document.getText();
     const lines = text.split('\n');
 
-    // get each line's possible API key warnings
-    // then flatten nested diagnostics into a flat array
-    const fileDiagnostics: Diagnostic[] = lines
-      .map(this.prepareLineDiagnostics)
-      .reduce((acc, next) => acc.concat([...next]), []);
+    const message = await this.git.isGitRepo(document.uri) ? diagnosticMessageGit : diagnosticMessageNoGit;
+
+    const fileDiagnostics: Diagnostic[] = lines.flatMap(this.prepareLineDiagnostics(message));
```

Comment:
Is it okay to remove the reduce? Are the diagnostics not actually nested?

Reply:
Originally we were mapping over the array, then flattening the result with `reduce`. Those two methods together should be equivalent to `flatMap`.

Reply:
Oh! I totally missed the `map` to `flatMap. My mistake